src/WebJobs.Extensions.Http/ClaimsPrincipalBindingProvider.cs

```diff
@@ -0,0 +1,116 @@
+    /// </summary>
+    internal class ClaimsPrincipalBindingProvider : IBindingProvider
+    {
+        private static readonly Task<IBinding> NullBinding = Task.FromResult<IBinding>(null);
+
+        private static readonly IReadOnlyCollection<string> PotentialIdentityHeaders = new HashSet<string>
+        {
+            "x-ms-client-principal", 
+            "x-ms-functions-key-identity"
+        };
+
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            if (context.Parameter.ParameterType != typeof(ClaimsPrincipal))
+            {
+                return NullBinding;
+            }
+
+            return Task.FromResult<IBinding>(new ClaimsIdentityBinding(context.Parameter));
+        }
+
+        private class ClaimsIdentityBinding : IBinding
+        {
+            private readonly ParameterInfo _parameter;
+
+            public ClaimsIdentityBinding(ParameterInfo parameter)
+            {
+                _parameter = parameter;
+            }
+
+            public bool FromAttribute
+            {
+                get { return false; }
+            }
+
+            public Task<IValueProvider> BindAsync(BindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                IReadOnlyDictionary<string, object> bindingData = context.BindingData;
+                ClaimsPrincipal principal = GetClaimsPrincipalFromBindingData(bindingData);
+
+                var valueProvider = new SimpleValueProvider(typeof(ClaimsPrincipal), principal, principal?.Identity?.Name);
+                return Task.FromResult<IValueProvider>(valueProvider);
+            }
+
+            private static ClaimsPrincipal GetClaimsPrincipalFromBindingData(IReadOnlyDictionary<string, object> bindingData)
+            {
+                object headers;
+                if (bindingData.TryGetValue(HttpTriggerAttributeBindingProvider.HttpHeadersKey, out headers))
+                {
+                    List<ClaimsIdentity> identities = (headers as IDictionary<string, string>)
+                        .Where(header => PotentialIdentityHeaders.Any(idHeader => string.Equals(idHeader, header.Key, StringComparison.OrdinalIgnoreCase)))
+                        .Select(header => ClaimsIdentityHelper.FromBase64EncodedJson(header.Value))
+                        .ToList();
+                    return new ClaimsPrincipal(identities);
+                }
+                else
+                {
+                    return new ClaimsPrincipal();
+                }
+            }
+
+            public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+            {
+                var request = value as ClaimsPrincipal;
```

Comment:
>request [](start = 20, length = 7)

Why is this variable named `request` and not `principal`? `request` sounds like an HTTP request, and that confused me when I was trying to understand what this code was doing.

Reply:
Not sure what happened here. Likely was taking inspiration from HttpTriggerAttributeBindingProvider and forgot to change variable name.

---
In reply to: [181944940](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/404#discussion_r181944940) [](ancestors = 181944940)